src/Workspaces/Core/Portable/Rename/RenameUtilities.cs

```diff
@@ -79,9 +79,8 @@ internal static IEnumerable<Document> GetDocumentsAffectedByRename(ISymbol symbo
                     .Concat(documentsOfRenameSymbolDeclaration.First().Id)
                     .Select(d => d.ProjectId).Distinct();
 
-                if (symbol.DeclaredAccessibility == Accessibility.Private)
+                if (ShouldRenameOnlyAffectDeclaringProject(symbol))
```

Comment:
I think I might want to additionally check "isSubset" and then include the extra projects if it is false rather than assert/throw, since it's much more likely that the rename cascaded from private to public than for us to have errantly found unrelated projects (I hope). But I'm not sure whether we'd rather keep it this way to keep the perf scenarios tightly monitored. @rchande - any thoughts?